src/main/scala/vsys/account/ContractAccount.scala

```diff
@@ -0,0 +1,115 @@
+package vsys.account
+
+import com.wavesplatform.state2.ByteStr
+import com.wavesplatform.utils.base58Length
+import scorex.account.AddressScheme
+import scorex.crypto.encode.Base58
+import scorex.crypto.hash.SecureCryptographicHash._
+import scorex.transaction.ValidationError
+import scorex.transaction.ValidationError.{InvalidContractAddress, InvalidAddress}
+import scorex.utils.ScorexLogging
+
+import scala.util.Success
+
+sealed trait ContractAccount extends Serializable {
+
+  val bytes: ByteStr
+  lazy val address: String = bytes.base58
+  lazy val stringRepr: String = address
+
+  override def toString: String = stringRepr
+
+  override def equals(obj: scala.Any): Boolean = obj match {
+    case conAcc: ContractAccount => bytes == conAcc.bytes
+    case _ => false
+  }
+
+  override def hashCode(): Int = java.util.Arrays.hashCode(bytes.arr)
+
+}
+
+object ContractAccount extends ScorexLogging {
+
+  val Prefix: String = "contractAccount:"
+
+  val AddressVersion: Byte = 6
+  val TokenAddressVersion: Byte = -124
```

Comment:
really? -124?

Reply:
This Byte will make tokenId.base58 to start with letter 'T'. This is the only reason.

Reply:
👍